user_sync/connector/directory_ldap.py

```diff
@@ -269,35 +272,55 @@ def iter_users(self, users_filter):
                 if (last_attribute_name != None):
                     self.logger.warn('No email attribute: %s for dn: %s', last_attribute_name, dn)
                 continue
+
+            source_attributes = {}
             
             user = user_sync.connector.helper.create_blank_user()
             user['email'] = email
+            source_attributes['email'] = email
                 
             username, last_attribute_name = self.user_username_formatter.generate_value(record)
             if (username == None and last_attribute_name != None):
                 self.logger.info('No username attribute: %s for dn: %s', last_attribute_name, dn)    
             user['username'] = username if username != None else email
+            if (username != None):
+                source_attributes['username'] = username
                     
             domain, last_attribute_name = self.user_domain_formatter.generate_value(record)
             if (domain != None):
                 user['domain'] = domain
+                source_attributes['domain'] = domain
             elif (last_attribute_name != None):
                 self.logger.info('No domain attribute: %s for dn: %s', last_attribute_name, dn)    
                                                 
             given_name_value = LDAPValueFormatter.get_attribute_value(record, 'givenName')
             if (given_name_value != None):   
                 user['firstname'] = given_name_value
+                source_attributes['givenName'] = given_name_value
             sn_value = LDAPValueFormatter.get_attribute_value(record, 'sn')
             if sn_value != None:
                 user['lastname'] = sn_value
+                source_attributes['sn'] = sn_value
             c_value = LDAPValueFormatter.get_attribute_value(record, 'c')
             if c_value != None:
                 user['country'] = c_value
+                source_attributes['c'] = c_value
                 
             uid = LDAPValueFormatter.get_attribute_value(record, 'uid')
             if (uid != None):
                 user['uid'] = uid
-            
+                source_attributes['uid'] = uid
+
+            if extended_attributes is not None:
+                for extended_attribute in extended_attributes:
+                    extended_attribute_value = LDAPValueFormatter.get_attribute_value(record, extended_attribute)
+                    if (extended_attribute_value is not None):
```

Comment:
This looks to be brittle if any of the base attributes we look for are also specified in the extended attributes, because we will overwrite the base attributes (which have special logic applied to them) with the raw value.  Are you checking to make sure there is no overlap?

Also this is another argument for the change I suggested at line 299, if we make the source_attributes for the Adobe-relevant values have the Adobe-relevant names, then if the code really wants the value of the `gn` attribute in the source it can specify `gn` as an extended attribute and get it separately direct from the source.